tests/v2/codecoverage/setuprancher/main.go

```diff
@@ -0,0 +1,511 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/pkg/api/scheme"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/tests/framework/clients/corral"
+	"github.com/rancher/rancher/tests/framework/clients/dynamic"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeapi/workloads/deployments"
+	nodepools "github.com/rancher/rancher/tests/framework/extensions/rke1/nodepools"
+	aws "github.com/rancher/rancher/tests/framework/extensions/rke1/nodetemplates/aws"
+	"github.com/rancher/rancher/tests/framework/extensions/token"
+	"github.com/rancher/rancher/tests/framework/extensions/unstructured"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	passwordgenerator "github.com/rancher/rancher/tests/framework/extensions/users/passwordgenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	appv1 "k8s.io/api/apps/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sUnstructured "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	adminPassword = os.Getenv("ADMIN_PASSWORD")
+)
+
+const (
+	corralName       = "ranchertestcoverage"
+	rancherTestImage = "ranchertest/rancher:v2.6-head"
+	namespace        = "cattle-system"
+	deploymentName   = "rancher"
+	clusterName      = "local"
+	// The json/yaml config key for the corral package to be build ..
+	userClusterConfigsConfigurationFileKey = "userClusterConfig"
+)
+
+type UserClusterConfig struct {
```

Comment:
why add a separate yaml definition for a new config here instead of updating the existing `CATTLE_CONFIG_TEST` ?

Reply:
Because I'm keeping track of three clusters along with the user tokens used to provision this cluster. This is to glue the python and go code together.

Reply:
Can't the existing config get updated with this instead of creating a new one ? I'm still not see the reasoning behind creating a new file.

Reply:
I'll try to explain better sorry. So we need three clusters to test our python, and go tests. In other words go validations, python v3_need_cluster tests and v3_additional_tests. The latter needs an admin token, user token, and cluster name to run. This config is keep track of what are my clusters I created and the tokens used to create them. We only need this because of the python tests. Does that make sense?